pkg/config/vpn-client.go

```diff
@@ -5,40 +5,47 @@
 package config
 
 import (
+	"fmt"
+	"slices"
 	"strconv"
 	"strings"
 	"time"
 
 	"github.com/caarlos0/env/v10"
+
+	"github.com/gardener/vpn2/pkg/constants"
 	"github.com/gardener/vpn2/pkg/network"
 )
 
 type VPNClient struct {
 	TCP struct {
-		KeepAliveTime     int64 `env:"KEEPALIVE_TIME" envDefault:"7200"`
-		KeepAliveInterval int64 `env:"KEEPALIVE_INTVL" envDefault:"75"`
-		KeepAliveProbes   int64 `env:"KEEPALIVE_PROBES" envDefault:"9"`
+		KeepAliveTime     uint64 `env:"KEEPALIVE_TIME" envDefault:"7200"`
+		KeepAliveInterval uint64 `env:"KEEPALIVE_INTVL" envDefault:"75"`
+		KeepAliveProbes   uint64 `env:"KEEPALIVE_PROBES" envDefault:"9"`
 	} `envPrefix:"TCP_"`
-	IPFamilies        string       `env:"IP_FAMILIES" envDefault:"IPv4"`
-	Endpoint          string       `env:"ENDPOINT"`
-	OpenVPNPort       int          `env:"OPENVPN_PORT" envDefault:"8132"`
-	VPNNetwork        network.CIDR `env:"VPN_NETWORK"`
-	SeedPodNetwork    network.CIDR `env:"SEED_POD_NETWORK"`
-	IsShootClient     bool         `env:"IS_SHOOT_CLIENT"`
-	PodName           string       `env:"POD_NAME"`
-	Namespace         string       `env:"NAMESPACE"`
-	VPNServerIndex    string       `env:"VPN_SERVER_INDEX"`
-	VPNClientIndex    int
-	IsHA              bool          `env:"IS_HA"`
-	ReversedVPNHeader string        `env:"REVERSED_VPN_HEADER" envDefault:"invalid-host"`
-	HAVPNClients      int           `env:"HA_VPN_CLIENTS"`
-	HAVPNServers      int           `env:"HA_VPN_SERVERS"`
-	PodLabelSelector  string        `env:"POD_LABEL_SELECTOR" envDefault:"app=kubernetes,role=apiserver"`
-	WaitTime          time.Duration `env:"WAIT_TIME" envDefault:"2s"`
+	IPFamilies           []string       `env:"IP_FAMILIES" envDefault:"IPv4"`
```

Comment:
Does the `envDefault` also work after the change to a slice or does that need to be adapted to something like `[]string{"IPv4"}`?

Reply:
providing a single value results in an array with a single element. there is also [a test](https://github.com/gardener/vpn2/pull/137/files#diff-f3b6280e440dae87302e39f882c1200e97275396df63deafd8b7792a552f6ddeR188) for it.